### ISSUE TITLE
Repeated execution of opts_chunk$set(fig.dim = ...) in {r setup} chunk causes error on notebook save

### ISSUE THREAD

**Issue by sz-cgt** [rstudio/rstudio#3450]
I originally filed this as a `knitr` issue (yihui/knitr#1600) but Yihui Xie suggests it's an RStudio issue instead

Running an `{r setup}` chunk that contains `knitr::opts_chunk$set(fig.dim = <vector>)` more than once in an R session produces an error message when saving the notebook, and breaks any preview etc.

To reproduce using the notebook below, 
1. run the `setup` chunk
2. save the workbook
3. run the `setup` chunk again
4. save the workbook again

The error displays on the second save

Two other observations:
1. The error does not occur is the R chunk is unnamed or named anything other than `setup`
2. The error does not occur when `fig.width` and  `fig.height` are used instead of `fig.dim`
````
---
title: "R Notebook"
output: html_notebook
---

```{r setup}
knitr::opts_chunk$set(fig.dim = c(3, 3))
```
````

|Field|Value|
|-----|-----|
|RStudio Edition|Desktop|
|RStudio Version|Version 1.1.456 
|OS Version|Windows 10 (x64)|
|R Version|R version 3.5.1 (2018-07-02)|

`sessionInfo()` prior to running chunk
> R version 3.5.1 (2018-07-02)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows >= 8 x64 (build 9200)
> Matrix products: default
> locale:
> [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
> [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
> [5] LC_TIME=English_Australia.1252    
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> loaded via a namespace (and not attached):
> [1] compiler_3.5.1  backports_1.1.2 rprojroot_1.3-2 tools_3.5.1    
> [5] yaml_2.2.0     

sessionInfo() after save error
> R version 3.5.1 (2018-07-02)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows >= 8 x64 (build 9200)
> Matrix products: default
> locale:
> [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
> [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
> [5] LC_TIME=English_Australia.1252    
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> loaded via a namespace (and not attached):
>  [1] compiler_3.5.1  backports_1.1.2 magrittr_1.5    rprojroot_1.3-2
>  [5] htmltools_0.3.6 tools_3.5.1     base64enc_0.1-3 yaml_2.2.0     
>  [9] Rcpp_0.12.18    stringi_1.1.7   rmarkdown_1.10  knitr_1.20     
> [13] jsonlite_1.5    stringr_1.3.1   digest_0.6.16   evaluate_0.11

**Comment by kevinushey** [comment]
The problem here is that somewhere, `fig.dim` sets changed from a numeric vector of length two to a list of length two, and `knitr` doesn't understand how to handle that. You can reproduce a similar issue with:

`````
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.dim = list(3, 3))
```
`````

This of course is RStudio's issue -- we shouldn't be doing this transformation. I'll see if we can figure out where that is happening.

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
